src/include/user/mfcc.h

```diff
@@ -0,0 +1,83 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+ */
+
+#ifndef __USER_MFCC_H__
+#define __USER_MFCC_H__
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#define MFCC_BLACKMAN_A0 Q_CONVERT_FLOAT(0.42, 15) /* For MFCC */
+
+#define SOF_MFCC_CONFIG_MAX_SIZE	256	/* Max size for configuration data in bytes */
+
+enum sof_mfcc_fft_pad_type {
+	MFCC_PAD_END = 0,
+	MFCC_PAD_CENTER = 1,
+	MFCC_PAD_START = 2,
+};
+
+enum sof_mfcc_fft_window_type {
+	MFCC_RECTANGULAR_WINDOW = 0,
+	MFCC_BLACKMAN_WINDOW = 1,
+	MFCC_HAMMING_WINDOW = 2,
+	MFCC_HANN_WINDOW = 3,
+	MFCC_POVEY_WINDOW = 4,
+};
+
+enum sof_mfcc_mel_log_type {
+	MEL_LOG_IS_LOG = 0,
+	MEL_LOG_IS_LOG10 = 1,
+	MEL_LOG_IS_DB = 2
+};
+
+enum sof_mfcc_mel_norm_type {
+	MFCC_MEL_NORM_NONE = 0,
+	MFCC_MEL_NORM_SLANEY = 1,
+};
+
+enum sof_mfcc_dct_type {
+	MFCC_DCT_I = 0,
+	MFCC_DCT_II = 1,
+};
+
+/*
+ * Configuration blob
+ */
+struct sof_mfcc_config {
+	int32_t sample_frequency; /**< Hz. e.g. 16000 */
```

Comment:
We need a size as first word for ABI tracking.

Reply:
That was a good catch! I added also some reserved words to begin. The three reserved bool in the end are to make the blob multiple of 32 bits size. The sof_setup() function now checks for size match also.